### PR TITLE
Business pages

### PR DESCRIPTION
**Pull request by DmytroDidukh**
- creating
- editing
- deleting
- editor
- upload images

### CODE REVIEW

**Review by maksym-strus — APPROVED** [review]

File: src/redux/businessPages/businessPages.sagas.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  }
+}
+
+function* handleBusinessPageUpdate({ payload }) {
+  const { id, newPage } = payload;
+  try {
+    yield put(setLoading(true));
+    yield call(updateBusinessPage, id, newPage);
+
+    const businessPages = yield call(getAllBusinessPages);
```
Comment (Alan18081) [inline src/redux/businessPages/businessPages.sagas.js:107]: There is no need to fetch again business pages after creation, you can just update in the state with update data of the business page and avoid unnecessary HTTP call

File: src/utils/use-business-handlers.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  const handleTabsChange = (event, newValue) => {
+    setTabsValue(newValue);
+  };
+
+  const handleCheckboxChange = (event) => {
+    setCheckboxes({ ...checkboxes, [event.target.name]: event.target.checked });
+  };
+
+  const languageCheckboxes = languages.map((lang, index) => (
```
Comment (Alan18081) [inline src/utils/use-business-handlers.js:35]: Please, avoid components inside hooks

File: src/utils/use-business-handlers.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    <FormControlLabel
+      key={index}
```
Comment (Alan18081) [inline src/utils/use-business-handlers.js:37]: Don't use `index` as a `key`, use some stable and unique properties like `lang`

File: src/utils/use-business-handlers.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      control={
+        <Checkbox
+          checked={checkboxes[`${lang}`]}
```
Comment (Alan18081) [inline src/utils/use-business-handlers.js:40]: Why do you use string interpolation `${lang}? Just use `checkboxes[lang]`

File: src/utils/use-business-handlers.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+          onChange={handleCheckboxChange}
+          name={`${lang}`}
```
Comment (Alan18081) [inline src/utils/use-business-handlers.js:42]: Why do you use string interpolation `${lang}? Just use `lang`

File: src/components/editor/file-blot/file-blot.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { Quill } from 'react-quill';
+
+const BlockEmbed = Quill.import('blots/block/embed');
+
+class FileBlot extends BlockEmbed {
+  static create(value) {
+    const prefixTag = document.createElement('span');
+    prefixTag.innerText = '첨부파일 - ';
```
[the remark this answers is not in the document]
Comment (DmytroDidukh) [reply src/components/editor/file-blot/file-blot.js:8]: it's from a Korean library

File: src/pages/business/business-add/business-add.js [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    setPreferredLanguages(prefLanguages);
+  }, [checkboxes, setPreferredLanguages]);
+
+  const langValues = languages.map((lang) => ({
+    [`${lang}Title`]: '',
+    [`${lang}Text`]: ''
+  }));
+
+  const formikValues =
+    langValues !== null ? Object.assign(...langValues) : null;
```
[the remark this answers is not in the document]
Comment (DmytroDidukh) [reply src/pages/business/business-add/business-add.js:53]: Yes, you are right, it's not necessary.

**Pull request merged**